protocols/relay/src/priv_client/handler.rs

```diff
@@ -340,74 +312,137 @@ impl ConnectionHandler for Handler {
+                limit,
+            }))) => {
+                let to_listener = self
+                    .active_reserve_requests
+                    .pop_front()
+                    .expect("must have active request for stream");
+
+                return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
+                    self.reservation.accepted(
+                        renewal_timeout,
+                        addrs,
+                        to_listener,
+                        self.local_peer_id,
+                        limit,
+                    ),
+                ));
+            }
+            Poll::Ready(Ok(Err(error))) => {
+                let mut to_listener = self
+                    .active_reserve_requests
+                    .pop_front()
+                    .expect("must have active request for stream");
+
+                self.send_channel_tasks.push(
+                    async move {
+                        let _ = to_listener
+                            .send(transport::ToListenerMsg::Reservation(Err(error)))
+                            .await;
+                    }
+                    .boxed(),
+                );
+                self.reservation.failed();
+            }
+            Poll::Ready(Err(futures_bounded::Timeout { .. })) => {
+                let mut to_listener = self
+                    .active_reserve_requests
+                    .pop_front()
+                    .expect("must have active request for stream");
+
+                self.send_channel_tasks.push(
+                    async move {
+                        let _ = to_listener
+                            .send(transport::ToListenerMsg::Reservation(Err(
+                                outbound_hop::ReserveError::Io(io::ErrorKind::TimedOut.into()),
+                            )))
+                            .await;
+                    }
+                    .boxed(),
+                );
+                self.reservation.failed();
             }
+            Poll::Pending => {}
+        }
+
+        debug_assert_eq!(
+            self.inflight_output_connect_requests.len(),
+            self.active_connect_requests.len(),
+            "expect to have one active request per inflight stream"
+        );
+
+        // Circuits
+        match self.inflight_output_connect_requests.poll_unpin(cx) {
+            Poll::Ready(Ok(Ok(outbound_hop::Circuit {
+                limit,
+                read_buffer,
+                stream,
+            }))) => {
+                let to_listener = self
+                    .active_connect_requests
+                    .pop_front()
+                    .expect("must have active request for stream");
+
+                let (tx, _) = oneshot::channel();
+
+                // TODO: What do we on error?
+                let _ = to_listener.send(Ok(priv_client::Connection {
+                    state: priv_client::ConnectionState::new_outbound(stream, read_buffer, tx),
+                }));
```

Comment:
Logging the error on debug should be enough, right? Apparently the listener is gone.